templates/ingress.yaml

```
@@ -31,10 +31,17 @@ spec:
       http:
         paths:
           {{- range .paths }}
+          {{- if kindIs "string" . }}
           - path: {{ . }}
             backend:
               serviceName: {{ $fullName }}
               servicePort: http
+          {{- else }}
+          - path: {{ .path }}
+            backend:
+              serviceName: {{ .serviceName | default $fullName }}
+              servicePort: {{ .servicePort | default "http" }}
```

How about:
```suggestion
          - {{ . | toYaml | indent 12 | trim }}
```

This way is still backwards compatible, but allows specifying arbitrary (future?) attributes, such as `pathType`. Admittedly you lose the nice default values that you supplied, though, but I think that's probably not a heavy price to pay.

---

Yes you're right, your approach is way better.
Thanks :+1: